app/core/auth.py

```diff
@@ -18,82 +17,68 @@
 )
 
 
-async def _get_jwt_user(token: str = Depends(security.oauth2_scheme)) -> JWTUser:
+def _decode_jwt(token: str = Depends(security.oauth2_scheme)) -> JWTUser:
     """Light-weight user-retrieval that only decodes the JWT."""
     try:
         payload = jwt.decode(
             token, security.SECRET_KEY, algorithms=[security.ALGORITHM]
         )
-        email: Optional[str] = payload.get("sub")
+
+        email: Optional[str] = payload.get("email")
         if email is None:
             raise CREDENTIALS_EXCEPTION
+
+        # TODO: Remove
         permissions: Optional[str] = payload.get("permissions")
         if permissions is None:
             raise CREDENTIALS_EXCEPTION
+
+        # TODO: Remove
         is_active: Optional[bool] = payload.get("is_active")
         if is_active is None:
             raise CREDENTIALS_EXCEPTION
 
+        authorization: Optional[dict[str, Any]] = payload.get("authorization")
+
         jwt_user = JWTUser(
             email=email,
             is_superuser=permissions == "admin",
             is_active=is_active,
+            authorization=authorization,
         )
         return jwt_user
     except PyJWTError:
         raise CREDENTIALS_EXCEPTION
 
 
-async def get_current_active_user(
```

Comment:
Assuming converting from async as we don't expect to have many authentication requests at all as users aren't authenticating?

Reply:
none of the code was actually async, or called in an async way, so I'm just cleaning up a potential issue here